TedTalks/Managers/TedTalkManager.swift

```diff
@@ -10,12 +10,41 @@ import Foundation
 class TedTalkManager {
     
     private(set) var tedTalks: [TedTalk] = []
+    private(set) var filteredTedTalks: [TedTalk] = []
     
     init() {
         self.tedTalks = Serialization.load(fromJSON: "ted") ?? []
+        self.filteredTedTalks = self.tedTalks
     }
     
     func add(tedTalk: TedTalk) {
         tedTalks.append(tedTalk)
     }
+    
+    func filter(keyword: String, filter: String) {
+        filteredTedTalks = keyword=="" ? tedTalks.filter{ doesTedTalkMatch(keyword, filter, $0)} : tedTalks
```

Comment:
Logic seems to be inverted here, in this case if `keyword` is an empty String it will try to perform the search to match that empty string, otherwise will default to the unfiltered list, so filtering is currently not working.

Reply:
Ok, fixed